src/connection/activate.ts

```diff
@@ -0,0 +1,94 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceEventName, WalletConnectionResult } from '@uniswap/analytics-events'
+import { sendEvent } from 'components/analytics'
+import { Connection } from 'connection'
+import { atom, useAtom } from 'jotai'
+import { useCallback } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import { updateSelectedWallet } from 'state/user/reducer'
+
+import { didUserReject } from './utils'
+
+type PendingConnectionState = {
+  connection: Connection | undefined
+  error: any
+}
+
+const pendingConnectionAtom = atom<PendingConnectionState>({ connection: undefined, error: undefined })
+
+export function useActivateConnection() {
+  const dispatch = useAppDispatch()
+
+  const [pending, setPending] = useAtom(pendingConnectionAtom)
```

Comment:
I would rename this from `pending` to something else, so it doesn't suggest the value is `boolean`, but an object. Maybe we could have two atoms, one for `connection` and another one for `error`. Also, any particular reason to use Jotai here instead of `useState`? AFAIK if we're not referring to this from the outside of this component, I don't think there's any reason for using that over state. But I may be wrong, so I am leaving this as an open question!

Reply:
Will rename 👍 I originally had it as two sep atoms, but it felt cleaner to combine into an object since we're generally updating them at the same time

These are moved into an atom because if this hook used `useState`, then both callers of this hook `<WalletModal />` and `<UniwalletModal />` would have their own pending connection state, when the desired functionality is for closing the `<UniwalletModal />` to update the same pending state that `<WalletModal />` uses

Reply:
Sounds good! Thanks for explaining that! If we have two call-sites, that definitely makes sense!